RHIDP-2405 : Added the content explaining vulnerabilities, SBOMs, and EC (#72)
redhat-developer/red-hat-developers-documentation-rhdh: modules/rhdh-plugins-reference/proc-setting-up-openshift-pipelines-to-download-or-view-sboms.adoc

:_mod-docs-content-type: PROCEDURE
[id="setting-up-openshift-pipelines-to-download-or-view-sboms_{context}"]

> = Setting up OpenShift Pipelines to view SBOMs

Reviewer: ```suggestion
= Setting up OpenShift Pipelines to view SBOM
```

Reviewer: There are few other places in other files which needed to be updated as well?

> In Red Hat Developer Hub, under the CI tab, the Pipeline Runs section offers a feature to view Software Bill of Material (SBOM). This enhances transparency and control over your supply chain by listing all the software libraries a component utilizes. These libraries may provide specific functionality or facilitate development.

Reviewer 2: Consider using the attribute (product for first occurence, product-short after):

```suggestion
In {product}, under the CI tab, the Pipeline Runs section offers a feature to view Software Bill of Material (SBOM). This enhances transparency and control over your supply chain by listing all the software libraries a component utilizes. These libraries may provide specific functionality or facilitate development.
```

> In Red Hat Developer Hub, under the CI tab, the Pipeline Runs section offers a feature to view Software Bill of Materials (SBOMs). This enhances transparency and control over your supply chain by listing all the software libraries a component utilizes. These libraries may provide specific functionality or facilitate development.

Reviewer: ```suggestion
In Red Hat Developer Hub, under the CI tab, the Pipeline Runs section offers a feature to view Software Bill of Materials (SBOM). This enhances transparency and control over your supply chain by listing all the software libraries a component utilizes. These libraries may provide specific functionality or facilitate development.
```

You can use an SBOM to better understand the composition of your software, identify vulnerabilities, and assess the potential impact of any security issues that might arise.

> .Options to download or view SBOMs

Reviewer: ```suggestion
.Options to download or view SBOM
```

image::rhdh-plugins-reference/sbom.png[]

[NOTE]
====
The SBOM task discussed in this chapter uses `syft` for example purposes only. `syft` is a community project and is not supported by Red Hat Developer Hub by default. For guidance on implementing these or any other community-supported tasks consult your support agreements or contact support services.
====

.Prerequisites

> * You have link:https://docs.openshift.com/container-platform/4.14/web_console/web-console.html#web-console[logged in to the web console].

Reviewer 2: Consider using the attribute:

```suggestion
* You have link:https://docs.openshift.com/container-platform/4.14/web_console/web-console.html#web-console[logged in to the {ocp-short} web console].
```

See the attribute list: https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/blob/main/artifacts/attributes.adoc

> * You have the appropriate link:https://docs.openshift.com/container-platform/4.14/authentication/using-rbac.html#default-roles_using-rbac[roles and permissions] in a project to create applications and other workloads in OpenShift Container Platform.

Reviewer 2: Consider using the attribute:

```suggestion
* You have the appropriate link:https://docs.openshift.com/container-platform/4.14/authentication/using-rbac.html#default-roles_using-rbac[roles and permissions] in a project to create applications and other workloads in {ocp-short}.
```

        echo 'BEGIN SBOM'
        cat sbom-image.json
        echo 'END SBOM'
        echo 'quay.io/user/workloads/<namespace>/node-express/node-express:build-8e536-1692702836' | tee $(results.LINK_TO_SBOM.path) # <8>
----
<1> The name of your task.
<2> The location for storing the task outputs.
<3> The SBOM task result name. Do not change the name of the SBOM result task.
<4> (Optional) Set to open the SBOM in a new tab.
<5> The description of the result.
<6> The image that generates the SBOM.
<7> The script that generates the SBOM image.
<8> The SBOM image along with the path name.

. Update the Pipeline to reference the newly created SBOM task.

+
[source,yaml]
----
...
spec:
  tasks:
    - name: sbom-task  
      taskRef: 
        name: sbom-task # <1>
  results:
    - name: IMAGE_URL  # <2>
      description: url
      value: <oci_image_registry_url> # <3>
----
<1> The same name as created in Step 2.
<2> The name of the result.
<3> The OCI image repository URL which contains the `.sbom` images.

. Rerun the modified OpenShift pipeline.

== Viewing an SBOM in the web UI

.Prerequisites

* You have set up {pipelines-shortname} to download or view SBOM.

.Procedure

> . Log in to Red Hat Developer Hub.

Reviewer 2: Consider using the attribute:

```suggestion
. Log in to {product}.
```
